core/controllers/reader.py

```diff
@@ -294,6 +298,69 @@ def get(self, exploration_id):
         self.render_json(self.values)
 
 
+class StorePlaythroughHandler(base.BaseHandler):
+    """Handles a useful playthrough coming in from the frontend to store it."""
+
+    @acl_decorators.can_play_exploration
+    def post(self, exploration_id):
+        """Handles POST requests. Appends to existing list of playthroughs or
+        deletes it if already full.
+
+        Args:
+            exploration_id: str. The ID of the exploration.
+        """
+        exp_issue_dict = self.payload.get('exp_issue')
+        playthrough_data = self.payload.get('playthrough_data')
+
+        exp_issues_model = stats_models.ExplorationIssuesModel.get(
+            exploration_id)
+        exp_issues = stats_services.get_exp_issues_from_model(exp_issues_model)
+
+        playthrough_id = stats_models.PlaythroughModel.create(
+            playthrough_data['exp_id'],
+            playthrough_data['exp_version'],
+            playthrough_data['issue_type'],
+            playthrough_data['issue_customization_args'],
+            playthrough_data['playthrough_actions'],
+            playthrough_data['is_valid'])
+
+        customization_args = exp_issue_dict['customization_args']
+
+        issue_found = False
+        for index, issue in enumerate(exp_issues.unresolved_issues):
+            if issue['issue_type'] == exp_issue_dict['issue_type']:
```

Comment:
Is there any guarantee that each issue in the list of unresolved issues has a different (type, key) tuple? If so, where is this enforced?

Reply:
@seanlip Whenever any issue is added to this list, it is done only if the combination of (type, key) doesnt already exist. Isn't that sufficient enough to know there are no duplicates?

Reply:
Yes as long as this is the only mechanism for adding new issues.